src/routes/members/gigs/[gig_id([0-9]+)]/edit.svelte

```diff
@@ -0,0 +1,1133 @@
+      this.error(404, "Gig not found");
+      return;
+    }
+
+    return {
+      ...gig,
+      lastSaved: gig,
+      venues,
+      gigTypes,
+      allContacts,
+    };
+  }
+</script>
+
+<script>
+  export let title,
+    date,
+    time,
+    posting_time,
+    posting_user,
+    editing_time,
+    editing_user,
+    lastSaved,
+    venues,
+    venue_id,
+    type_id,
+    gigTypes,
+    id,
+    admins_only,
+    advertise,
+    allow_signups,
+    food_provided,
+    notes_admin,
+    notes_band,
+    summary,
+    contacts,
+    allContacts,
+    arrive_time,
+    finish_time,
+    finance,
+    finance_deposit_received,
+    finance_payment_received,
+    finance_caller_paid,
+    quote_date;
+  import { makeTitle, themeName } from "../../../../view";
+  import Select from "../../../../components/Forms/Select.svelte";
+  import VenueEditor from "../../../../components/Gigs/VenueEditor.svelte";
+  import ContactEditor from "../../../../components/Gigs/ContactEditor.svelte";
+  import Summary from "../../../../components/Gigs/Summary.svelte";
+  import Fuse from "fuse.js";
+  import { client as graphqlClient } from "../../../../graphql/client";
+  import { tick } from "svelte";
+  import { tweened } from "svelte/motion"; // TODO move the messages to their own component (#34)
+  import { fade } from "svelte/transition";
+  import { stores } from "@sapper/app";
+  import SearchBox from "../../../../components/SearchBox.svelte";
+  import { DateTime, Settings } from "luxon";
+  Settings.defaultZoneName = "Europe/London"; // https://moment.github.io/luxon/docs/manual/zones#changing-the-default-zone
+
+  const { session } = stores();
+
+  let venue;
+  let displayVenueEditor = false;
+  let displayContactEditor = false;
+  let editingSubvenue = false;
+  let recentlySavedOpacity = tweened(0, { duration: 150 });
+  let recentlySavedTimer = () => recentlySavedOpacity.set(0);
+  let venueListElement,
+    clientListElement,
+    callerListElement,
+    selectedClient,
+    selectedCaller,
+    contactToEdit,
+    editContactType,
+    clearVenueSearch;
+  let previewSummary = false;
+  time = time && DateTime.fromISO(`1970-01-01T${time}`).toFormat("HH:mm"); // Remove seconds from time if they are present
+  let arrive_time_date = (arrive_time && DateTime.fromISO(arrive_time).toFormat("yyyy-LL-dd")) || null;
+  let arrive_time_time = (arrive_time && DateTime.fromISO(arrive_time).toFormat("HH:mm")) || null;
+  let finish_time_date = (finish_time && DateTime.fromISO(finish_time).toFormat("yyyy-LL-dd")) || null;
+  let finish_time_time = (finish_time && DateTime.fromISO(finish_time).toFormat("HH:mm")) || null;
+  let fields = {};
+  $: timingWarnings = [
+    arrive_time &&
+      date &&
+      time &&
+      DateTime.fromISO(arrive_time) <= DateTime.fromISO(`${date}T${time}`).minus({ hours: 3 }) &&
+      "Arrive time is 3 hours or more before the start time. Have you accidentally put a time in the morning rather than the evening?",
```

Comment:
is it worth making these warnings dismissable somehow? or rare enough edge case that this is intentional that we don't care

Reply:
Going with rare enough I don't care since dismissing them would be a lot of effort for little gain since the vast majority of gigs will be fine with these